MREUnityRuntime/MREUnityRuntimeLib/Messaging/Commands/CommandHandlerContext.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace MixedRealityExtension.Messaging.Commands
+{
+	/// <summary>
+	/// This class pre-caches method info for command handlers, since it does not change at runtime.
+	/// This saved us 20ms on the startup frame for the MREApp on a high end CPU.
+	/// </summary>
+	/// <typeparam name="T"></typeparam>
+	internal class CommandHandlerContext<T> : ICommandHandlerContext
```

Comment:
>CommandHandlerContext [](start = 16, length = 21)

This is interesting. @Tombu what do you think? We were discussing the removal of exactly this reflection earlier today. Look like a good solution?

Reply:
It's still doing reflection here. just during the static constructor of the first instance of the object.  I still think that there is room to improve this further by removing the need for reflection altogether.

---
In reply to: [419882530](https://github.com/microsoft/mixed-reality-extension-unity/pull/147#discussion_r419882530) [](ancestors = 419882530)

Reply:
Even so, it's now a one-time startup cost. That's a significant improvement.

---
In reply to: [421122983](https://github.com/microsoft/mixed-reality-extension-unity/pull/147#discussion_r421122983) [](ancestors = 421122983,419882530)